src/components/settings/Settings.tsx

```diff
@@ -1,39 +1,71 @@
+import TerminalSettings from "./Terminal";
+import "./Settings.css";
 
 export interface SettingsProps {
-    onClose: () => void
+  onClose: () => void;
 }
 
-const MENU_BACKGROUND = 'background';
-const MENU_QUOTE = 'quote';
-const MENU_TERMINAL = 'terminal';
+enum MenuItem {
+  Background = "background",
+  Quote = "quote",
+  Terminal = "terminal",
+}
+
+interface Menu {
+  name: string;
+  type: MenuItem;
+  elem: React.ReactElement<any, any>;
+}
+
+const Menus: Menu[] = [
+  {
+    name: "Background",
+    type: MenuItem.Background,
+    elem: <BackgroundSettings />,
+  },
+  { name: "Quote", type: MenuItem.Quote, elem: <QuoteSettings /> },
+  { name: "Terminal", type: MenuItem.Terminal, elem: <TerminalSettings /> },
+];
 
 export default function Settings(props: SettingsProps) {
-    const { onClose } = props;
-    const [menu, setMenu] = useState(MENU_BACKGROUND);
-    return (
-        <div className="settings">
-            <div className="close-button" onClick={() => onClose()}>
-                <img alt="cross" src="/assets/cross.svg" width="20px" height="20px" />
-            </div>
-            <div className="navbar">
-                <div className={`menu ${menu === MENU_BACKGROUND ? 'active' : ''}`} onClick={() => setMenu(MENU_BACKGROUND)}>Background</div>
-                <div className={`menu ${menu === MENU_QUOTE ? 'active' : ''}`} onClick={() => setMenu(MENU_QUOTE)}>Quote</div>
-                <div className={`menu ${menu === MENU_TERMINAL ? 'active' : ''}`} onClick={() => setMenu(MENU_TERMINAL)}>Terminal</div>
-            </div>
-            <div className="divider" />
-            <div className="settings-content">
-                {menu === MENU_BACKGROUND && <BackgroundSettings />}
-                {menu === MENU_QUOTE && <QuoteSettings />}
-                {menu === MENU_TERMINAL && <TerminalSettings />}
-            </div>
+  const { onClose } = props;
+  const [menu, setMenu] = useState(MenuItem.Background);
+
+  return (
+    <div className="settings">
+      <div className="close-button" onClick={() => onClose()}>
+        <img alt="cross" src="/assets/cross.svg" width="20px" height="20px" />
+      </div>
+      <div className="navbar">
+        {Menus.map((item) =>
+          createMenuItem(menu, setMenu, item.type, item.name)
+        )}
+        <div className="menu" onClick={() => {}}>
+          Restore Defaults
         </div>
-    )
+      </div>
+      <div className="divider" />
+      <div className="settings-content">
+        {Menus.filter((item) => menu === item.type).map((item) => item.elem)}
+      </div>
+    </div>
+  );
 }
 
-const TerminalSettings = () => {
-    return <div></div>
-}
+const createMenuItem = (
+  currentItem: MenuItem,
+  setMenu: (value: any) => void,
+  menuItem: MenuItem,
+  menuName: string
+): React.ReactElement<any, any> => (
+  <div
+    className={`menu ${currentItem === menuItem ? "active" : ""}`}
+    onClick={() => setMenu(menuItem)}
+  >
+    {menuName}
```

Comment:
minor: I prefer to have custom component <MenuItem> here instead of a function, then rename:
* `currentItem` => `isActive`
* `setMenu` => `onClick`
* `menuItem` => can be removed